webpack/components/OpenscapRemediationWizard/steps/ReviewHosts.js

```diff
@@ -0,0 +1,198 @@
+import { useBulkSelect } from 'foremanReact/components/PF4/TableIndexPage/Table/TableHooks';
+import { getPageStats } from 'foremanReact/components/PF4/TableIndexPage/Table/helpers';
+import { getControllerSearchProps, STATUS } from 'foremanReact/constants';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+
+import OpenscapRemediationWizardContext from '../OpenscapRemediationWizardContext';
+import WizardHeader from '../WizardHeader';
+import {
+  HOSTS_API_PATH,
+  HOSTS_API_REQUEST_KEY,
+  FAIL_RULE_SEARCH,
+} from '../constants';
+
+const ReviewHosts = () => {
+  const { source, hostId, setHostIdsParam } = useContext(
+    OpenscapRemediationWizardContext
+  );
+
+  const defaultSearch = `${FAIL_RULE_SEARCH} = ${source}`;
+  const defaultParams = {
+    search: defaultSearch,
+  };
+
+  const [params, setParams] = useState(defaultParams);
+
+  const searchProps = getControllerSearchProps('hosts');
+  searchProps.autocomplete.searchQuery = defaultSearch;
+  searchProps.autocomplete.url = '../../hosts/auto_complete_search'; // TODO: find a way to not use relative path
+  searchProps.disabled = true; // This is to force hosts to be searched by currently remediated rule only
+
+  const response = useAPI('get', `${HOSTS_API_PATH}?include_permissions=true`, {
+    key: HOSTS_API_REQUEST_KEY,
+    params: defaultParams,
+  });
+  const {
+    response: {
+      search: apiSearchQuery,
+      results,
+      per_page: perPage,
+      page,
+      subtotal,
+      message: errorMessage,
+    },
+    status = STATUS.PENDING,
+    setAPIOptions,
+  } = response;
+
+  const subtotalCount = Number(subtotal ?? 0);
+
+  const setParamsAndAPI = newParams => {
+    setParams(newParams);
+    setAPIOptions({ key: HOSTS_API_REQUEST_KEY, params: newParams });
+  };
+
+  const onSearch = newSearch => {
+    if (newSearch !== apiSearchQuery) {
+      setParamsAndAPI({ ...params, search: newSearch, page: 1 });
+    }
+  };
+
+  const { pageRowCount } = getPageStats({
+    total: subtotalCount,
+    page,
+    perPage,
+  });
+  const { fetchBulkParams, ...selectAllOptions } = useBulkSelect({
+    results,
+    metadata: { total: subtotalCount, page },
+    initialSearchQuery: apiSearchQuery || defaultSearch,
+    isSelectable: () => true,
+    initialArry: [hostId],
+  });
+  const {
+    selectPage,
+    selectedCount,
+    selectOne,
+    areAllRowsOnPageSelected,
+    areAllRowsSelected,
+    isSelected,
+  } = selectAllOptions;
+
+  useEffect(() => {
+    if (selectedCount) setHostIdsParam(fetchBulkParams());
+  }, [selectedCount, fetchBulkParams, setHostIdsParam]);
+
+  const selectionToolbar = (
+    <ToolbarItem key="selectAll">
+      <SelectAllCheckbox
+        {...{
+          selectAll: noop, // I don't think it really can select all since ids from other pages are still need to be loaded/fetched
+          selectPage,
+          selectNone: noop, // we shouldn't allow to select none since it migth lead to selecting a wrong host
```

Comment:
Select none should select the host the button was clicked on, of have some kind of action

Reply:
I've tried that with calling imported selectNone and then using selectOne as part of selectNone, but for some reason it didn't want to select the host, so I've decided to hide Select None option as with Select All. That'd need a patch in Foreman though...

Reply:
I tried:
``` js
selectNone: () => {
            selectNone();
            selectOne(true, hostId);
          },
```
and it worked fine, still its a bit confusing that the main checkbox doesn't select the whole page, but I'm sure if its a fix for the plugin or foreman core

Reply:
That's just weird... Probably the new way of testing UI is through incognito mode :/ Never had issues with something being cached before. Or maybe that was something else. I've just checked again and it seems to be working indeed.

Reply:
Jeremey said that he had this issue, and he checked the disable cache box in the dev tools, network tab and it got resolved, maybe it will help? (I have it on for years so I never notice these issues ): )